### PR TITLE
Update XMLs & Add script for analytical results of the EDP wellbore example

### PR DESCRIPTION
**Pull request by sytuannguyen**
This PR address the issue #2436 with following tasks:

- Update the xml benchmark file to pass the running error,
- Add `PackCollection`, `timeHistory`, etc. to output `hdf5` data for validation,
- Add vtk output,
- Add python script for the analytical results that are used for validating GEOS results,
- Change GEOSX to GEOS in the legends,
- Update the doc for edp wellbore example.

### CODE REVIEW

**Review by herve-gross — APPROVED** [review]

File: inputFiles/solidMechanics/ExtendedDruckerPragerWellbore_benchmark.xml [hunk 1]
```diff
@@ -10,32 +10,57 @@
...
       zCoords="{ -1, 1 }"
       nr="{ 40 }"
-      nt="{ 80 }"
-      nz="{ 2 }"
+      nt="{ 40 }"
+      nz="{ 1 }"
       trajectory="{ { 0.0, 0.0, -1.0 },
                     { 0.0, 0.0, 1.0 } }"
       autoSpaceRadialElems="{ 1 }"
-      useCartesianOuterBoundary="0"
```
Comment (jhuang2601) [inline inputFiles/solidMechanics/ExtendedDruckerPragerWellbore_benchmark.xml:22]: I did not see the reason why switching to radial shaped outer boundary. I prefer to use the Cartesian aligned outer boundary.

File: inputFiles/solidMechanics/ExtendedDruckerPragerWellbore_benchmark.xml [hunk 2]
```diff
@@ -10,32 +10,57 @@
...
-
   <!-- SPHINX_DP_WELLBORE_MESH_END -->
   
   <!--Solver application-->
   <Events
-    maxTime="1.0">
+    maxTime="1.05">
     <PeriodicEvent
       name="solverApplications"
+      forceDt="0.5"
```
Comment (jhuang2601) [inline inputFiles/solidMechanics/ExtendedDruckerPragerWellbore_benchmark.xml:31]: It is better to lower the forceDt here to capture the wellbore shape in the elastic regime.

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/Example.rst [hunk 3]
```diff
@@ -180,38 +177,30 @@ The parameters used in the simulation are summarized in the following table.
...
 +------------------+-------------------------+------------------+---------------+
-| :math:`C`        | Cohesion                | [MPa]            | 0.0           |
+| :math:`c`        | Cohesion                | [MPa]            | 0.0           |
 +------------------+-------------------------+------------------+---------------+
 | :math:`\phi_i`   | Initial Friction Angle  | [degree]         | 15.27         |
 +------------------+-------------------------+------------------+---------------+
 | :math:`\phi_r`   | Residual Friction Angle | [degree]         | 23.05         |
 +------------------+-------------------------+------------------+---------------+
-| :math:`c_h`      | Hardening Rate          | [-]              | 0.01          |
+| :math:`m`      | Hardening Rate          | [-]              | 0.01          |
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/Example.rst:186]:
```suggestion
| :math:`m`        | Hardening Rate          | [-]              | 0.01          |
```

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/Example.rst [hunk 4]
```diff
@@ -138,8 +135,8 @@ The next step is to specify fields, including:
   - The boundary conditions (the reduction of wellbore pressure and constraints of the outer boundaries have to be set)
 
 In this example, we need to specify isotropic horizontal stress (:math:`\sigma_h` = -11.25 MPa) and vertical stress (:math:`\sigma_v` = -15.0 MPa). 
-To reach equilibrium, a compressive traction :math:`P_w` = -11.25 MPa is instantaneously applied at the wellbore wall ``rneg`` at time :math:`t` = 0 s, which will then be gradually reduced to a lower value (-2.0 MPa) to let wellbore contract.
-The remaining parts of the outer boundaries are subjected to roller constraints.  
+To reach equilibrium, a compressive traction :math:`P_w` = -11.25 MPa is instantaneously applied at the wellbore wall ``rneg`` at time :math:`t` = 0 s, which will then be gradually reduced to a lower absolute value (-2.0 MPa) to let wellbore contract.
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/Example.rst:138]:
```suggestion
To reach equilibrium, a compressive traction :math:`p_w` = -11.25 MPa is instantaneously applied at the wellbore wall ``rneg`` at time :math:`t` = 0 s, which will then be gradually reduced to a lower absolute value (-2.0 MPa) to let wellbore contract.
```

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/displacementHistory.hdf5
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/displacementHistory.hdf5:1]: This hdf5 file is too large to be saved on the develop repo. Please remove it and keep the related python scripts and generated figure.

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/stressHistory_rock.hdf5
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/stressHistory_rock.hdf5:1]: See above

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/Example.rst [hunk 5]
```diff
@@ -180,38 +177,30 @@ The parameters used in the simulation are summarized in the following table.
...
-.. _problemVerificationEDPWellboreFig:
-.. figure:: Verification.png
-   :align: center
-   :width: 1000
-   :figclass: align-center
-
-   Comparing GEOS results with analytical solutions
+In the above example, we requested hdf5 output files. We can therefore use python scripts to visualize the outcome. Below figure shows the comparisons between the numerical predictions (marks) and the corresponding analytical solutions (solid curves) with respect to the distributions of principal stress components, stress path on the wellbore surface, the supporting wellbore pressure and wellbore size. It is clear that the GEOS predictions are in excellent agreement with the analytical results. On the top-right figure, we added also a comparison between GEOS results for elasto-plastic material and the anlytical solutions of an elastic material. Note that the elastic solutions are differed from the elasto-plastic results even in the elastic zone (r/a>2).
 
+.. plot:: docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/Example.rst:203]: Please also introduce this python script, when mentioning the locations of xml files at the beginning.

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+		list_p_wellsurface_analytic.append(p_wellsurface_analytic)
+		list_q_wellsurface_analytic.append(q_wellsurface_analytic)
+
+	# Plots	
+	plt.figure(figsize=(15,10))
+
+	# Plot GEOS results versus analytical results for elasto-plastic material
+	plt.subplot(2,2,1)
+	plt.plot( rCoord/rCoord[0],-stress_rr,'r+',label=r'$\sigma_{rr}$: GEOS')
+	plt.plot(r_ep_analytic,srr_ep_analytic, 'r', label=r'$\sigma_{rr}$: Analytic Plastic')
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:165]:
```suggestion
	plt.plot(r_ep_analytic,srr_ep_analytic, 'r', label=r'$\sigma_{rr}$: Analytical Plasticity')
```

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+	plt.plot(r_elas_analytic,srr_elas_analytic, 'r')
+
+	plt.plot( rCoord/rCoord[0],-stress_tt,'b+',label=r'$\sigma_{\theta\theta}$: GEOS')
+	plt.plot(r_ep_analytic,stt_ep_analytic, 'b', label=r'$\sigma_{\theta\theta}$: Analytic Plastic')
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:169]:
```suggestion
	plt.plot(r_ep_analytic,stt_ep_analytic, 'b', label=r'$\sigma_{\theta\theta}$: Analytical Plasticity')
```

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+	plt.plot(r_elas_analytic,stt_elas_analytic, 'b')
+
+	plt.plot( rCoord/rCoord[0],-stress_zz,'g+',label=r'$\sigma_{zz}$: GEOS')
+	plt.plot(r_ep_analytic,szz_ep_analytic, 'g', label=r'$\sigma_{zz}$: Analytic Plastic')
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:173]:
```suggestion
	plt.plot(r_ep_analytic,szz_ep_analytic, 'g', label=r'$\sigma_{zz}$: Analytical Plasticity')
```

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+	plt.yticks(np.linspace(0.0,20e6,11))
+	plt.xscale('log', subs=range(2,10))
+	plt.legend()
+
+	# Plot GEOS results for elasto-plastic material versus analytical results for elastic material
+	r_elas_assumed,srr_elas_assumed,stt_elas_assumed,szz_elas_assumed = epwAnal.solution_elastic(sh,sv,1.0,pw)
+
+	plt.subplot(2,2,2)
+	plt.plot( rCoord/rCoord[0],-stress_rr,'r+',label=r'$\sigma_{rr}$: GEOS')
+	plt.plot(r_elas_assumed,srr_elas_assumed, 'r', label=r'$\sigma_{rr}$: Analytic Elastic')
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:191]:
```suggestion
	plt.plot(r_elas_assumed,srr_elas_assumed, 'r', label=r'$\sigma_{rr}$: Analytical Elasticity')
```

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+
+	plt.plot( rCoord/rCoord[0],-stress_tt,'b+',label=r'$\sigma_{\theta\theta}$: GEOS')
+	plt.plot(r_elas_assumed,stt_elas_assumed, 'b', label=r'$\sigma_{\theta\theta}$: Analytic Elastic')
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:194]:
```suggestion
	plt.plot(r_elas_assumed,stt_elas_assumed, 'b', label=r'$\sigma_{\theta\theta}$: Analytical Elasticity')
```

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+
+	plt.plot( rCoord/rCoord[0],-stress_zz,'g+',label=r'$\sigma_{zz}$: GEOS')
+	plt.plot(r_elas_assumed,szz_elas_assumed, 'g', label=r'$\sigma_{zz}$: Analytic Elastic')
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:197]:
```suggestion
	plt.plot(r_elas_assumed,szz_elas_assumed, 'g', label=r'$\sigma_{zz}$: Analytical Elasticity')
```

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+	finalFrictionAngle *= np.pi/180.0 # converted to rad
+	param_b_i = edpAnal.compute_param_b(initialFrictionAngle)
+	param_b_f = edpAnal.compute_param_b(finalFrictionAngle)
+	p_yieldSurface = np.arange(0,15e6,1)
+	q_yieldSurface_i = p_yieldSurface*param_b_i
+	q_yieldSurface_f = p_yieldSurface*param_b_f
+
+	plt.subplot(2,2,3)
+	plt.plot(-p,q,'ko',label='GEOS')
+	plt.plot(list_p_wellsurface_analytic,list_q_wellsurface_analytic,'b',label='Analytic')
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:218]:
```suggestion
	plt.plot(list_p_wellsurface_analytic,list_q_wellsurface_analytic,'b',label='Analytical')
```

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+	plt.xlim(0,15e6)
+	plt.ylim(0,10e6)
+	plt.legend()
+
+	# Plot the wellbore deformation a0/a versus wellbore surface pressure pw
+	pw = stress_xx_wellboresurface
+
+	plt.subplot(2,2,4)
+	plt.plot(a0/a, -pw,'ko',label='GEOS')
+	plt.plot(list_a0_a_ratio,list_pw_analytic,'b',label='Analytic')
```
Comment (jhuang2601) [inline src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:233]:
```suggestion
	plt.plot(list_a0_a_ratio,list_pw_analytic,'b',label='Analytical')
```

**Review by jhuang2601 — CHANGES_REQUESTED** [review]
Thanks for updating the example with timeHistory functions!

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/Example.rst (hunk 3 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/Example.rst:186]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/Example.rst (hunk 4 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/Example.rst:138]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/displacementHistory.hdf5
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/displacementHistory.hdf5:1]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/stressHistory_rock.hdf5
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/stressHistory_rock.hdf5:1]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/Example.rst (hunk 5 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/Example.rst:203]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py (hunk 6 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:165]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py (hunk 7 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:169]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py (hunk 8 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:173]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py (hunk 9 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:191]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py (hunk 10 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:194]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py (hunk 11 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:197]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py (hunk 12 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:218]: done

File: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py (hunk 13 above)
Comment (sytuannguyen) [reply src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/edpWellbore/edpWellbore_plot.py:233]: done

File: inputFiles/solidMechanics/ExtendedDruckerPragerWellbore_benchmark.xml (hunk 1 above)
Comment (sytuannguyen) [reply inputFiles/solidMechanics/ExtendedDruckerPragerWellbore_benchmark.xml:22]: The shape of the far-field boundary does not really matter in this example. However a cylindrical shape of the far-field boundary provides a more optimized local mesh around the wellbore surface for this axisymmetric problem.

File: inputFiles/solidMechanics/ExtendedDruckerPragerWellbore_benchmark.xml (hunk 2 above)
Comment (sytuannguyen) [reply inputFiles/solidMechanics/ExtendedDruckerPragerWellbore_benchmark.xml:31]: Done. The number of step in the elastic regime was reduced to minimize the size of the hdf5 outputs. Now the hdf5 are removed, the time step in the elastic regime can be reduced.

**Pull request merged**